client-course-schedulizer/src/readCSV.ts

```diff
@@ -0,0 +1,428 @@
+              [, hourPart, minPart] = regMatch;
+              numHourPart = Number(hourPart);
+
+              // Handle high hour values
+              if (numHourPart > 11) {
+                if (numHourPart > 12) {
+                  // If military time, convert to standard
+                  numHourPart -= 12;
+                  hourPart = String(numHourPart);
+                }
+                // Assume PM when 12:XX or military time
+                ampm = "PM";
+              }
+
+              // If hour is 0, assume military time of 12 AM
+              if (numHourPart === 0) {
+                hourPart = "12";
+                numHourPart = 12;
+              }
+
+              // Look to see whether AM or PM is specified explicitly
+              if (pmReg.test(value)) {
+                ampm = "PM";
+                if (amReg.test(value)) {
+                  // eslint-disable-next-line no-console
+                  console.log(`Time of "${value}" is labeled with AM and PM, defaulting to PM`);
+                }
+              } else if (amReg.test(value)) {
+                ampm = "AM";
+              }
+
+              // Piece the time together
+              startTime = `${hourPart}:${minPart} ${ampm}`;
+            } else {
+              // eslint-disable-next-line no-console
+              console.log(`Time of "${value}" is unreadable, defaulting to 8:00 AM`);
+            }
+            break;
+          }
+          case "duration": {
+            duration = Number(value);
+            break;
+          }
+          case "location": {
+            roomParts = value.trim().split(" ");
+            if (roomParts.length === 1) {
+              // No room number given
+              [building] = roomParts;
+              roomNumber = "";
+            } else if (roomParts.length === 2) {
+              // Building and room number given
+              [building, roomNumber] = roomParts;
+            } else {
+              // Too many room parts given, assume last part is room number and rest is building
+              building = roomParts.slice(0, -1).join(" ");
+              [roomNumber] = roomParts.slice(-1);
+            }
+            break;
+          }
+          case "roomCapacity": {
+            roomCapacity = Number(value);
+            break;
+          }
+          case "year": {
+            year = Number(value);
+            break;
+          }
+          case "term": {
+            if (fallReg.test(value)) {
+              term = di.Term.Fall;
+            } else if (summerReg.test(value)) {
+              term = di.Term.Summer;
+            } else if (springReg.test(value)) {
+              term = di.Term.Spring;
+            } else if (interimReg.test(value)) {
+              term = di.Term.Interim;
+            } else {
+              // eslint-disable-next-line no-console
+              console.log(`Term of "${value}" is unreadable, defaulting to Fall`);
```

Comment:
Repeated structure like this within code usually signifies that it can be refactored into a function that is more flexible. For example, you could in the case call a function that tests the 'value' and when it is true it assigns 'term = di.Term[term]' where term is calculated by the function based on when regular expression resulted to true. This provides more flexible code. For example, if we were to get another term called Winter, then all you would have to do is add a new RegularExpression for Winter and update the Interface and the function would take care of the rest. (This is kinda a dumb example because the terms are likely to be only these four, but this thinking can be applied to the "half", and "days" below. But since it is working, maybe we just leave it as is?)

Reply:
Tried doing something like this for a while, but had some difficulty with it, will just leave it as is for now.